platform/posix/clock_posix.c

```diff
@@ -63,8 +67,9 @@ void Clock_SleepMs( uint32_t sleepTimeMs )
 {
     /* Convert parameter to timespec. */
     struct timespec sleepTime = { 0 };
-    sleepTime.tv_sec = sleepTimeMs / MILLISECONDS_PER_SECOND;
-    sleepTime.tv_nsec = ( sleepTimeMs % MILLISECONDS_PER_SECOND ) * NANOSECONDS_PER_MILLISECOND;
+
+    sleepTime.tv_sec = ( ( time_t ) sleepTimeMs / ( time_t ) MILLISECONDS_PER_SECOND );
+    sleepTime.tv_nsec = ( ( int64_t ) sleepTimeMs % MILLISECONDS_PER_SECOND ) * NANOSECONDS_PER_MILLISECOND;
```

Comment:
Is the `int64_t` cast for just `sleepTimeMs` or for the `sleepTimeMs % MILLISECONDS_PER_SECOND` expression?

Reply:
As the POSIX specification uses `long` as the type for `tv_nsec`, should we use that type here instead?
https://pubs.opengroup.org/onlinepubs/007908799/xsh/time.h.html

Reply:
Use of long is also a MISRA directive violation. We cannot directly use that.
int64_t cast is for sleepTimeMs